src/sidebar/util/annotation-sharing.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const serviceConfig = require('../service-config');
+
+/**
+ * Is the sharing of annotations enabled? Check for any defined `serviceConfig`,
+ * but default to `true` if none found.
+ *
+ * @param {object} settings
+ * @return {boolean}
+ */
+function sharingEnabled(settings) {
+  const serviceConfig_ = serviceConfig(settings);
+  if (serviceConfig_ === null) {
+    return true;
+  }
+  if (typeof serviceConfig_.enableShareLinks !== 'boolean') {
+    return true;
+  }
+  return serviceConfig_.enableShareLinks;
+}
+
+/**
+ * Return any defined standalone URI for this `annotation`, preferably the
+ * `incontext` URI, but fallback to `html` link if not present.
+ *
+ * @param {object} annotation
+ * @return {string|undefined}
+ */
+function shareURI(annotation) {
```

Comment:
We currently have both `annotation-metadata` and `annotation-sharing` modules. All the functions in `annotation-metadata` take only an annotation as an argument and have no dependencies on the rest of the application.

Reply:
Also food for thought. I had considered putting at least some of this in the `metadata` util, but noticed that (to my taste, at least), the module was starting to feel a little long and cluttered. Granted, however, the functions in that module are simple and there aren't many dependencies (internal or external). So, yeah, worth a chat!

Reply:
> I had considered putting at least some of this in the metadata util, but noticed that (to my taste, at least), the module was starting to feel a little long and cluttered.

It is currently ~230 lines. I really don't think it is of a size where length alone should be a concern yet. The common feature of all code in `annotation-metadata.js` is that all the functions take only an annotation as input, so there is no dependency on app-specific concerns such as configuration etc. That is admittedly a difference from the helpers introduced here.